libqtile/widget/thermal_zone.py

```diff
@@ -0,0 +1,29 @@
+from libqtile.widget import base
+
+class ThermalZone(base.ThreadPoolText):
+    """
+    This is simple thermal zone monitoring widget.
+
+    Widget requirements: linux kernel >= 5.x.x
+
+    It allows you to get CPU temperature in Celsius degrees from /sys/class/thermal/ sensors.
+    """
+    orientations = base.ORIENTATION_HORIZONTAL
+    defaults = [
+        ('update_interval', 2.0, 'Update interval for the thermal zone sensor'),
+        ('zone', '0', 'Default thermal zone'),
+    ]
+
+    def __init__(self, **config):
+        super().__init__("", **config)
+        self.add_defaults(ThermalZone.defaults)
+
+    def get_temp(self):
+        with open('/sys/class/thermal/thermal_zone' + self.zone + '/temp') as f:
+            value = str(round(int(f.read().rstrip()) / 1000)) + '°C'
+        return value
+
+    def poll(self):
```

Comment:
You could just put your `get_temp` code in `poll`.

Reply:
Done